ui: Statements tracing - Integrate Activate Diagnostics modal in Statements page

koorosh opened the pull request:
Depends on: #45577

koorosh committed:
ui: (fix) Apply styles for navigation panes

Before, Text component styles were adjusted to style
links on side navigation panel. It was wrong because
Text component should not care about particular styling
of navigation pane.
Now, particular styles which relate to navigation panel
only is moved to SideNavigation component.
It is changed now because current implementation breaks
styles for Diagnostics panel.

Release note: None

koorosh committed:
ui: Link component

Link component is a wrapper on top of <a> element
is created to maintain consistent styles for links.

Release note: None

koorosh committed:
ui: Add Diagnostics view for Statements

- Adjust SummaryCard to provide external styles
- Add DiagnosticsView component which renders
view when no diagnostics available.
- Current implementation includes only view without
any logic connected to redux store.

Release note: None

koorosh committed:
ui: Add Diagnostics view for Statements

- Adjust SummaryCard to provide external styles
- Add DiagnosticsView component which renders
view when no diagnostics available.
- Current implementation includes only view without
any logic connected to redux store.

Release note: None

koorosh committed:
ui: Add Diagnostics view for Statements

- Adjust SummaryCard to provide external styles
- Add DiagnosticsView component which renders
view when no diagnostics available.
- Current implementation includes only view without
any logic connected to redux store.

Release note: None

koorosh committed:
ui: Fake endpoints for Statement tracing data

Note: this commit must not be merged in master branch
it is for demo purposes only.

It extends `getStatements` api response with diagnostics field
which contains tracing data info.

Release note: None

koorosh committed:
ui: Fake endpoints for Statement tracing data

Note: this commit must not be merged in master branch
it is for demo purposes only.

It extends `getStatements` api response with diagnostics field
which contains tracing data info.

Release note: None

koorosh committed:
ui: Add sagas to handle statement tracing actions

Statements sagas are implemented to be able request
statement diagnostics and retrieve when its ready.

This commit defines Actions for requesting diagnostics
and sagas which listens for those actions.

Also selectors are added to find statement by
statement id (query) which contains diagnostic data.

Release note: None

koorosh committed:
ui: Diagnostics tab with tracing records

- Add table to represent diagnostics data of statement
- Connect Statements view with statement selector to properly
display tab header (whether diagnostics data is present or not)

Release note (admin ui): Add Diagnostics tab for statement details page

koorosh committed:
ui: Diagnostics tab with tracing records

- Add table to represent diagnostics data of statement
- Connect Statements view with statement selector to properly
display tab header (whether diagnostics data is present or not)

Release note (admin ui): Add Diagnostics tab for statement details page

koorosh committed:
ui: Diagnostics tab with tracing records

- Add table to represent diagnostics data of statement
- Connect Statements view with statement selector to properly
display tab header (whether diagnostics data is present or not)

Release note (admin ui): Add Diagnostics tab for statement details page

koorosh committed:
ui: Diagnostics tab with tracing records

- Add table to represent diagnostics data of statement
- Connect Statements view with statement selector to properly
display tab header (whether diagnostics data is present or not)

Release note (admin ui): Add Diagnostics tab for statement details page

koorosh committed:
ui: Add tooltips for diagnostics statuses

- All code related to statements diagnostics are moved to
statements/diagnostics dir;
- Status badges are refactored to external component which is
responsible for displaying proper text of status and text for
tooltips.
- Tooltips have to be shown on some views and disabled on other.
To implement this, `enableTooltip` property is added. Particulary
it is needed to hide tooltip for "READY" status in Diagnostics tab
and show it on Statements List view.

Release note (admin ui): Show tooltips for statement diagnostic statuses

koorosh committed:
ui: Add tooltips for diagnostics statuses

- All code related to statements diagnostics are moved to
statements/diagnostics dir;
- Status badges are refactored to external component which is
responsible for displaying proper text of status and text for
tooltips.
- Tooltips have to be shown on some views and disabled on other.
To implement this, `enableTooltip` property is added. Particulary
it is needed to hide tooltip for "READY" status in Diagnostics tab
and show it on Statements List view.

Release note (admin ui change): Show tooltips for statement diagnostic statuses

koorosh committed:
ui: Add tooltips for diagnostics statuses

- All code related to statements diagnostics are moved to
statements/diagnostics dir;
- Status badges are refactored to external component which is
responsible for displaying proper text of status and text for
tooltips.
- Tooltips have to be shown on some views and disabled on other.
To implement this, `enableTooltip` property is added. Particulary
it is needed to hide tooltip for "READY" status in Diagnostics tab
and show it on Statements List view.

Release note (admin ui): Show tooltips for statement diagnostic statuses

koorosh committed:
ui: Create HeaderSection shared component

It is common for pages to have header with 'back' navigation link.
HeaderSection component wraps page header with logic of navigation
back to specified page via props.

Release note: None

koorosh committed:
ui: Extend Statements table with Diagnostics column

Current changes add column which doesn't consume any
data and always shows 'Activate' link.
It is starting point for next changes:
- providing diagnostics statuses;
- displaying modal for requesting diagnostics;

Release note (admin ui): Add Diagnostics column to Statements Page

koorosh committed:
ui: Add Modal component

- Modal component is a wrapper around Ant.design Modal component
with customized styles;

Release note: None

koorosh committed:
ui: Integrate ActivateDiagnosticsModal with Statements page

ActivateDiagnosticsModal is implemented with total encapsulation
of modal's state and connected props to store into component.
- ActivateDiagnosticsModal component instance exposes `showModalFor`
method through forwardedRef that makes possible to manage open/closed
state of modal inside of component and parent component can trigger
where modal should be opened and provide required arguments.
Current pattern is more anti-pattern but with this particular case
it make easy to separate responsibilities between components.
- ActivateDiagnosticsModal component is connected to store and
dispatches action to request new diagnostics;

Statements table is based on SortedTable component which doesn't
provide easy way to handle clicks on rows (due to some magic
bootstrapping process). This required to propagate method which
triggers modal opening through several functions outside of
component.

Release note (admin ui change): Open Activate Diagnostics modal from Statements page (Activate link)

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/45610)
<!-- Reviewable:end -->

dhartunian committed:
DELETE!!! add API for accessing statement diagnostics

Added 3 new RPC endpoints under `_status` for manipulating Statement Diagnostics
Requests:

1. `CreateStatementDiagnosticsRequest`
Creates a new request for a given statement fingerprint

2. `StatementDiagnosticsRequests`
Fetches all statement diagnostics requests

3. `StatementDiagnostics`
Fetches a specific statement diagnostics request that has been completed and
includes the JSON payload of the trace

Basic implementations for these endpoints have been added to the
`apiReducers.tsx` and `api.ts` files for use by the Admin UI.

A minor bug where completed requests did not have `completedAt` timestamp set
was fixed.

Release note (<category, see below>): <what> <show> <why>

koorosh committed:
ui: Rename Link component to Anchor

Release note: None

koorosh committed:
ui: DEL! Stab for Statement Diagnostics API

Release note: None

koorosh committed:
ui: Integrate requestStatementDiagnostics API

Replace faked API with real implementation to
request statement diagnostics.

Release note: None

koorosh committed:
ui: Add selectDiagnosticRequestsByStatementFingerprint selector

Release note: None

koorosh committed:
ui: Connect Statement Diagnostics API

Before, Statements diagnostics view used faked API
to simulate responses from server.
Now redux API has changed to comply with actual
implementation of API.

Release note: None

koorosh committed:
ui: Connect Statement Diagnostics API

Before, Statements diagnostics view used faked API
to simulate responses from server.
Now redux API has changed to comply with actual
implementation of API.

Release note: None

koorosh committed:
ui: Connect Statement Diagnostics API wit diagnostics table component

Replacing faked API responses with real implementation required some
redesigns with following important changes:

- Requesting diagnosticsRequests initiated when DiagnosticsView is opened.
Then to request new diagnostics requests before existing data is invalidated,
we need to manually invalidate and then refresh data.

- Current backend API doesn't support Error state for requested data so only
"Ready", and "Waiting for query" statuses are possible.

Release note: None

koorosh committed:
ui: Add DownloadFile component

DownloadFile component accepts file name and content and
then initiates file download from client.
It works as a wrapper to catch onClick event on wrapped
component.

Release note: None

koorosh committed:
ui: Refactor Statement diagnostics action ()redux) names

Release note: None

koorosh committed:
ui: Add traces downloading on Statement Diagnostics tab

- Add ability to download json file with tracing data
- Refactor code to comply with changed backend API
- add `long` package because exposed interface for
getting statement diagnostics require diagnostics id
as Long type. Current library provides compatible
implementation of Long type with protobuf.js interfaces.

Release note: None

koorosh committed:
DEL! ui: Extend Table component with pagination size props

commit b0ab7def584ff8e6e1a22eb9129e4b72dcb4de0d
Author: Andrii Vorobiov <and.vorobiov@gmail.com>
Date:   Mon Mar 2 15:38:08 2020 +0200

    ui: Extend Table component with pagination size props

    Allow passing pageSize prop to Table component to
    define how many row should be displayed per page.

    Release note: None

Release note (<category, see below>): <what> <show> <why>

koorosh committed:
DEL! ui: Extend Text component with 'code' type

commit c4733bff62c39186a92f0112d9134f665e4f1df4
Author: Andrii Vorobiov <and.vorobiov@gmail.com>
Date:   Mon Mar 2 15:29:50 2020 +0200

    ui: Extend Text component with 'code' type

    Add styles to support monospace text by Text component.

    Release note: None

Release note (<category, see below>): <what> <show> <why>

koorosh committed:
ui: Add Statement Diagnostics History page

- Page contains list of all available statement diagnostics reports
- Added link on Debug page to reference on Statement Diagnostics History page
- Added link on Statement Diagnostics tab to reference on Statement Diagnostics History page

Release note (admin ui change): Statement Diagnostics History page added

koorosh commented:
Outdated
